components/logins/src/db.rs

```diff
@@ -489,6 +507,7 @@ impl LoginDb {
         }
         let tx = self.unchecked_transaction()?;
         let now_ms = util::system_time_ms_i64(SystemTime::now());
+        let import_start = now_ms;
```

Comment:
IIUC, `now_ms` needs to use the actual system time because we write it into the db. For measuring duration we should use [`std::time::Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html) rather than `std::time::SystemTime`, which has the advantage of guaranteeing a monotonic clock.

The API for this is kind of unusual, but really well designed for doing the sort of timekeeping you're doing here, something like:

```
let import_start = std::time::Instant::now();
...do some stuff...
fixup_phase_end = std::time::Instant::now();
...do some more stuff...
insert_phase_end = std::time::Instant::now();

fixup_phase_duration = fixup_phase_end.saturating_duration_since(import_start).as_millis();
import_phase_duration = import_phase_end.saturating_duration_since(fixup_phase_end).as_millis();
```

This will also avoid the concern with negative durations that you mention below.